src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -328,6 +358,27 @@ private TypeSymbol NarrowArrayAssignmentType(TypeValidatorConfig config, ArraySy
             return new TypedArrayType(TypeHelper.CreateTypeUnion(arrayProperties), targetType.ValidationFlags);
         }
 
+        private TypeSymbol NarrowLambdaType(TypeValidatorConfig config, LambdaSyntax lambdaSyntax, LambdaType targetType)
+        {
+            var returnType = NarrowType(config, lambdaSyntax.Body, targetType.ReturnType.Type);
+
+            var variables = lambdaSyntax.GetLocalVariables().ToImmutableArray();
+            if (variables.Length != targetType.ArgumentTypes.Length)
+            {
+                // TODO error message
```

Comment:
>// TODO error message [](http://example.com/codeflow?start=16&length=21)

Is the TODO still valid? If not, how does this condition lead to `AnyTypeIsNotAllowed()`?

Reply:
Yes it was :)